library/core/src/main/java/com/google/android/exoplayer2/upstream/ContentDataSource.java

```diff
@@ -71,7 +71,7 @@ public long open(DataSpec dataSpec) throws ContentDataSourceException {
     try {
       uri = dataSpec.uri;
       assetFileDescriptor = resolver.openAssetFileDescriptor(uri, "r");
-      inputStream = new FileInputStream(assetFileDescriptor.getFileDescriptor());
+      inputStream = assetFileDescriptor.createInputStream();
```

Comment:
Rather than doing this, I think it would be preferable to modify L84 to do:
```
bytesRemaining = assetFileDescriptor.getLength();
if (bytesRemaining == AssetFileDescriptor.UNKNOWN_LENGTH) {
  // The asset must extend to the end of the file.
  bytesRemaining = inputStream.available();
  if (bytesRemaining == 0) {
    // FileInputStream.available() returns 0 if the remaining length cannot be determined, or
    // if it's greater than Integer.MAX_VALUE. We don't know the true length in either case,
    // so treat as unbounded.
    bytesRemaining = C.LENGTH_UNSET;
  }
}
```
Which makes it explicit what's going on (and also allows bytesRemaining to be greater than Integer.MAX_VALUE, at least in theory).